pkg/urbit/vere/io/fore.c

```diff
@@ -69,6 +69,32 @@ _fore_inject(u3_auto* car_u, c3_c* pax_c)
   u3z(ovo);
 }
 
+/* _fore_import(): form an ovum from jammed archive at [pax_c] and inject it.
+*/
+static void
+_fore_import(u3_auto* car_u, c3_c* pax_c)
+{
+  // With apologies
+  u3_noun arc = u3ke_cue(u3m_file(pax_c));
+  c3_c * b64_c = u3r_string(u3do("crip", u3do("en-base64:mimes:html", arc)));
+  c3_w siz = strlen(b64_c) + 120;
+
+  c3_c bod_c[siz];
+  snprintf(bod_c, siz,
+    "{\"source\": {\"import-all\": {\"base64-jam\": \"%s\"}}, \
+    \"sink\": {\"stdout\": null}}", b64_c);
+  u3_noun dat = u3nt(u3_nul, u3i_word(strlen(bod_c)), u3i_string(bod_c));
+
+  u3_noun req = u3nt(c3n,
+    u3nc(u3i_string("ipv4"), u3i_word(0x7f000001)),
+    u3nq(u3i_string("POST"), u3i_string("/"), u3_nul, dat));
+  u3_noun wir = u3nc(u3i_string("http-server"), u3_nul);
+  u3_noun cad = u3nc(u3i_string("request-local"), req);
+  u3_auto_peer(
+    u3_auto_plan(car_u, u3_ovum_init(0, c3__e, wir, cad)),
+    0, 0, _fore_inject_bail);
```

Comment:
`_fore_inject_bail()` is the error callback here -- you'll probably want to define your own so that you can at least print a clear message if the import fails. If that should be a fatal error, you can call `u3_pier_bail(car_u->pir_u);`.